lib/util/StackedSetMap.js

```diff
@@ -5,24 +5,72 @@
 
 "use strict";
 
-const TOMBSTONE = {};
-const UNDEFINED_MARKER = {};
+const TOMBSTONE = Symbol("tombstone");
+const UNDEFINED_MARKER = Symbol("undefined");
+
+/**
+ * @template T
+ * @typedef {T | true | undefined} Cell<T>
+ */
+
+/**
+ * @template T
+ * @typedef {T | true | typeof TOMBSTONE | typeof UNDEFINED_MARKER} InternalCell<T>
+ */
+
+/**
+ * @template K
+ * @template V
+ * @param {[K, InternalCell<V>]} pair the internal cell
+ * @returns {[K, Cell<V>]} its “safe” representation
+ */
+const extractPair = pair => {
+	const key = pair[0];
+	const val = pair[1];
+	if (val === UNDEFINED_MARKER || val === TOMBSTONE) {
```

Comment:
It's impossible to have a `TOMBSTONE` here, as `_compress` removes all tombstones.

Reply:
True. But type refinement fails if I remove the test.